src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.WriteRaw.cs

```diff
@@ -0,0 +1,506 @@
+            }
+        }
+
+        public static IEnumerable<object[]> GetRootLevelPrimitives()
+        {
+            Action<byte[]> validate;
+
+            validate = (data) => Assert.Equal(123456789, JsonSerializer.Deserialize<long>(data));
+            yield return new object[] { Encoding.UTF8.GetBytes("123456789"), validate };
+
+            validate = (data) => Assert.Equal(1234.56789, JsonSerializer.Deserialize<double>(data));
+            yield return new object[] { Encoding.UTF8.GetBytes("1234.56789"), validate };
+
+            validate = (data) => Assert.Equal(@"Hello", JsonSerializer.Deserialize<string>(data));
+            yield return new object[] { Encoding.UTF8.GetBytes(@"""Hello"""), validate };
+
+            validate = (data) => Assert.Equal(s_guid, JsonSerializer.Deserialize<Guid>(data));
+            byte[] guidAsJson = WrapInQuotes(Encoding.UTF8.GetBytes(TestGuidAsStr));
+            yield return new object[] { guidAsJson, validate };
+        }
+
+        public static IEnumerable<object[]> GetArrays()
+        {
+            Action<byte[]> validate;
+
+            byte[] json = JsonSerializer.SerializeToUtf8Bytes(Enumerable.Repeat(1234.56789, 4));
+            validate = (data) =>
+            {
+                foreach (double d in JsonSerializer.Deserialize<double[]>(data))
+                {
+                    Assert.Equal(1234.56789, d);
+                }
+            };
+            yield return new object[] { json, validate };
+
+            json = JsonSerializer.SerializeToUtf8Bytes(Enumerable.Repeat("Hello", 4));
+            validate = (data) =>
+            {
+                foreach (string str in JsonSerializer.Deserialize<string[]>(data))
+                {
+                    Assert.Equal("Hello", str);
+                }
+            };
+            yield return new object[] { json, validate };
+
+            json = JsonSerializer.SerializeToUtf8Bytes(Enumerable.Repeat("Hello", 4));
+            validate = (data) =>
+            {
+                foreach (string str in JsonSerializer.Deserialize<string[]>(data))
+                {
+                    Assert.Equal("Hello", str);
+                }
+            };
+            yield return new object[] { json, validate };
+        }
+
+        private static byte[] WrapInQuotes(ReadOnlySpan<byte> buffer)
+        {
+            byte[] quotedBuffer = new byte[buffer.Length + 2];
+            quotedBuffer[0] = (byte)'"';
+            buffer.CopyTo(quotedBuffer.AsSpan(1));
+            quotedBuffer[buffer.Length + 1] = (byte)'"';
+            return quotedBuffer;
+        }
+
+        [Theory]
+        [InlineData(true, 0, "[]")]
+        [InlineData(false, 0, "[]")]
+        [InlineData(true, 1, "[1]")]
+        [InlineData(false, 1, "[1]")]
+        [InlineData(true, 5, "[1,1,1,1,1]")]
+        [InlineData(false, 5, "[1,1,1,1,1]")]
+        public static void WriteRawArrayElements(bool skipInputValidation, int numElements, string expectedJson)
```

Comment:
Add user-defined whitespace \ formatting? (here and elsewhere). That was one of the reasons for the API.

e.g.
```InlineData(true, 5, "[ 1,  1,1,1,1 ] ")]```

Reply:
Thanks! Added some whitespace tests